tkg/client/update_credentials.go

```diff
@@ -197,3 +210,118 @@ func (c *TkgClient) updateVSphereCredentialsForCluster(clusterClient clusterclie
 
 	return nil
 }
+
+// UpdateAzureClusterCredentials update azure cluster credentials
+func (c *TkgClient) UpdateAzureClusterCredentials(clusterClient clusterclient.Client, options *UpdateCredentialsOptions) error {
+	if options.AzureUpdateClusterOptions.AzureTenantID == "" || options.AzureUpdateClusterOptions.AzureClientID == "" || options.AzureUpdateClusterOptions.AzureClientSecret == "" {
+		return errors.New("either tenantId, clientId or clientSecret should not be empty")
+	}
+
+	if !options.IsRegionalCluster {
+		unifiedIdentity, err := clusterClient.CheckUnifiedAzureClusterIdentity(options.ClusterName, options.Namespace)
+		if err != nil {
+			return err
+		}
+		if unifiedIdentity {
+			log.Warningf("AzureCluster %s use the same AzureClusterIdentity from its management cluster. It cannot be updated separately.", options.ClusterName)
+			return nil
+		}
+	}
+
+	if err := c.UpdateAzureCredentialsForCluster(clusterClient, options, false); err != nil {
+		return err
+	}
+
+	if options.IsRegionalCluster {
+		clusters, err := clusterClient.ListClusters("")
+		if err != nil {
+			return errors.Wrapf(err, "unable to list workload clusters")
+		}
+
+		for i := range clusters {
+			if clusters[i].Name == options.ClusterName {
+				continue
+			}
+			unifiedIdentity, err := clusterClient.CheckUnifiedAzureClusterIdentity(clusters[i].Name, clusters[i].Namespace)
+			if err != nil {
+				log.Error(err, "unable to update credentials for workload cluster")
+				continue
```

Comment:
Better to return error here?

Reply:
One cluster under the same MC should not affect other clusters and MC.

Reply:
I am wondering if this will hide some failures and cause issues later?

Reply:
This is a good point that we may check only when the cluster is ready, we will allow users to update credentials? While CheckUnifiedAzureClusterIdentity seems not decide if the cluster is alive or not.

Reply:
Now it returns error simply. And the problem is tracked as an issue.